source/common/http/conn_manager_utility.cc

```diff
@@ -183,6 +183,13 @@ ConnectionManagerUtility::MutateRequestHeadersResult ConnectionManagerUtility::m
                                                                 : Headers::get().SchemeValues.Http);
   }
 
+  // Usually, the x-forwarded-port header comes with x-forwarded-proto header. Only set the
+  // x-forwarded-port header if configured and it is not already set.
+  if (config.appendXForwardedPort() && !request_headers.ForwardedPort()) {
+    request_headers.setForwardedPort(
+        connection.streamInfo().downstreamAddressProvider().localAddress()->ip()->port());
```

Comment:
I think we need to check on localAddress existing.  AFIK it could be a file descriptor

Reply:
Fixed, thanks.